### PR TITLE
Retain dedup clean for switchers

### PR DESCRIPTION
**Pull request by MarionJHolloway**
# Description
Previously all data was removed from a location which changed between care home and non-res. This PR updates the logic so that ascwds_filled_posts_dedup_clean is used for these locations, if available.

# How to test
Unit tests passing
Run in branch: https://eu-west-2.console.aws.amazon.com/states/home?region=eu-west-2#/v2/executions/details/arn:aws:states:eu-west-2:344210435447:execution:retain-dedup-clean-for-switche-Ind-CQC-Filled-Post-Estimates-Pipeline:f29291a1-8100-441d-80f9-86414ea8a0a4

# Developer checklist
- [x] Unit test
- [x] Linked to Trello ticket: https://trello.com/c/z77bbABj/747-for-places-who-switch-between-nr-and-ch-copy-over-ascwdsfilled-postsdeduplicatedclean
- [x] Documentation up to date

### CODE REVIEW

File: tests/test_file_data.py [hunk 1]
```diff
@@ -3455,24 +3455,24 @@ class EstimateMissingAscwdsFilledPostsData:
...
+        ("loc 1", date(2024, 1, 1), CareHome.care_home, 20.0, 10.0),
+        ("loc 1", date(2024, 2, 1), CareHome.care_home, 20.0, 10.0),
+        ("loc 2", date(2024, 1, 1), CareHome.not_care_home, 20.0, 10.0),
+        ("loc 2", date(2024, 2, 1), CareHome.not_care_home, 20.0, 10.0),
+        ("loc 3", date(2024, 1, 1), CareHome.care_home, 20.0, None),
+        ("loc 3", date(2024, 2, 1), CareHome.not_care_home, 20.0, None),
+        ("loc 4", date(2024, 1, 1), CareHome.not_care_home, 20.0, None),
+        ("loc 4", date(2024, 2, 1), CareHome.care_home, 20.0, 10.0),
     ]
     expected_null_changing_carehome_status_rows = [
```
Comment (garypolzin) [inline tests/test_file_data.py:3467]: might need to split these tests into two now - test names imply they are expecting nulls, but only some of them are null now

**Review by garypolzin — CHANGES_REQUESTED** [review]
think the tests need a tweak to match the new outcome but the job itself looks perfect

**Review by garypolzin — APPROVED** [review]
